gpg.cpp

```diff
@@ -69,7 +69,9 @@ std::string gpg_get_uid (const std::string& fingerprint)
 	command.push_back(gpg_get_executable());
 	command.push_back("--batch");
 	command.push_back("--with-colons");
-	command.push_back("--fixed-list-mode");
+	if (gpg_get_executable() != "gpgsm") {
```

Comment:
fixed-list-mode is the default mode since gpg 2.0.10 per https://www.gnupg.org/documentation/manuals/gnupg/GPG-Input-and-Output.html#index-fixed_002dlist_002dmode
gpgsm takes its version number from the version of the gnupg it's packaged with so maybe a version check would be better here? That way you could drop this option for gpg versions that don't need it. That might not work for alternative pgp binaries like smimesign though.